### PR TITLE
[SVS-5] Fix issues with adding Additional Files

### PR DESCRIPTION
**Pull request by mjcheetham**
@duncanpMS pointed out two issues, corrected both with these changes.
Issue 1: `AdditionalFiles` not `AdditionalFile` => renamed.
Issue 2: Files are not shown in Solution Explorer immediately => use EnvDTE object model rather than MSBuild.

### CODE REVIEW

File: src/Integration/Constants.cs [hunk 1]
```diff
@@ -27,11 +27,16 @@ public static class Constants
         /// <summary>
         /// The property key which corresponds to the Roslyn analyzer additional files
         /// </summary>
-        public const string AdditionalFilePropertyKey = "AdditionalFile";
+        public const string AdditionalFilesPropertyKey = "AdditionalFiles";
```
Comment (duncanpMS) [inline src/Integration/Constants.cs]: Strictly speaking this isn't a property: "AdditionalFilesItemTypeName" / "AdditionalFilesItemTypeKey" would be better.

Comment (mjcheetham) [inline src/Integration/Constants.cs]: Renamed to `AdditionalFilesItemTypeName`

File: src/Integration.UnitTests/Binding/BindingWorkflowTests.cs [hunk 2]
```diff
@@ -385,49 +384,12 @@ public void BindingWorkflow_InjectAdditionalFilesIntoProjects()
...
-            // Act
-            testSubject.InjectAdditionalFilesIntoProjects();
-
-            // Verify
-            VerifyAdditionalFileAddedToProject(project1, msbuildProject1, additionalFilePath1);
-            VerifyAdditionalFileAddedToProject(project2, msbuildProject2, additionalFilePath1);
-        }
-
-        [TestMethod]
-        public void BindingWorkflow_InjectAdditionalFilesIntoProjects_ExistingFile_DoesNotAddToProject()
```
Comment (duncanpMS) [inline src/Integration.UnitTests/Binding/BindingWorkflowTests.cs]: Is it no longer worth testing this case?

File: src/TestInfrastructure/Framework/ConfigurableVsProjectSystemHelper.cs [hunk 3]
```diff
@@ -57,19 +55,23 @@ void IProjectSystemHelper.AddFileToProject(Project project, string file)
...
-        Microsoft.Build.Evaluation.Project IProjectSystemHelper.GetEquivalentMSBuildProject(EnvDTE.Project project)
+        void IProjectSystemHelper.AddFileToProject(Project project, string file, string itemType)
         {
-            if (this.MsBuildProjectMapping.ContainsKey(project))
+            bool addFileToProject = !project.ProjectItems.OfType<ProjectItem>().Any(pi => StringComparer.OrdinalIgnoreCase.Equals(pi.Name, file));
+            if (addFileToProject)
             {
-                return this.MsBuildProjectMapping[project];
+                var item = project.ProjectItems.AddFromFile(file);
+                Property itemTypeProperty = item.Properties.Cast<Property>().FirstOrDefault(x => x.Name == Constants.ItemTypePropertyKey);
```
Comment (eyalantzMS) [inline src/TestInfrastructure/Framework/ConfigurableVsProjectSystemHelper.cs]: itemType

File: src/Integration.UnitTests/Binding/BindingWorkflowTests.cs (hunk 2 above)
Comment (mjcheetham) [inline src/Integration.UnitTests/Binding/BindingWorkflowTests.cs]: Not at this level. The `ProjectSystemHelper` is handling the check for already existing files whereas previously the InjectAdditionalFiles method was doing work to resolve whether or not to add the file.

**Comment by eyalantzMS** [comment]
LGTM

File: src/Integration/Binding/BindingWorkflow.cs [hunk 4]
```diff
@@ -385,28 +384,9 @@ private void InjectAdditionalFiles()
...
             foreach (var project in this.projectSystemHelper.GetSolutionManagedProjects())
             {
-                string projectRoot = PathHelper.ForceDirectoryEnding(Path.GetDirectoryName(project.FullName));
-
-                MSBuild.Project msBuildProject = this.projectSystemHelper.GetEquivalentMSBuildProject(project);
-
-                Debug.Assert(msBuildProject != null, "Couldn't find equivalent MSBuild project object to VS project object");
-
-                if (msBuildProject != null)
+                foreach (var additionalFilePath in this.AdditionalFilePaths)
```
Comment (duncanpMS) [inline src/Integration/Binding/BindingWorkflow.cs]: Minor optimisation, unrelated to the bug: there's no point in walking the projects in the solution if there are no additional files. The calling method _InjectAdditionalFiles_ could early out if there are no files to add.

Comment (mjcheetham) [inline src/Integration/Binding/BindingWorkflow.cs]: Done.

**Comment by mjcheetham** [comment]
Rebased off of `master deb556a4f06dd3bf3880cb58106c1c3a0858a095`

**Pull request merged**